pkg/frontend/frontend.go

```diff
@@ -152,6 +166,94 @@ func NewFrontend(ctx context.Context,
 	return f, nil
 }
 
+// NewKubernetesClient is a wrapper for restconfig.RestConfig and kubernetes.NewForConfig
+// which returns (kubernetes.Interface, error) instead of (*kubernetes.Clientset, error),
+// which allows it to be used as a kubernetesClientFactory
+func NewKubernetesClient(args ...interface{}) (kubernetes.Interface, error) {
+	if len(args) != 2 {
+		return nil, errors.New("Required arguments: (env.Interface, *api.OpenShiftCluster)")
+	}
+	env, ok := args[0].(env.Interface)
+	if !ok {
+		return nil, errors.New("First argument must be of type env.Interface")
+
+	}
+	oc, ok := args[1].(*api.OpenShiftCluster)
+	if !ok {
+		return nil, errors.New("Second argument must be of type *api.OpenShiftCluster")
+	}
+	restConfig, err := restconfig.RestConfig(env, oc)
+	if err != nil {
+		return nil, err
+	}
+	return kubernetes.NewForConfig(restConfig)
+}
+
+// Option allows kubeactions to only initialize the fields
+// required by a route handler
+type Option func(f *frontend)
+
+// WithKubeActionsFactory sets a KubeActions factory in frontend
+func WithKubeActionsFactory(factory kubeActionsFactory) Option {
+	return func(f *frontend) {
+		f.kubeActionsFactory = factory
+	}
+}
+
+// WithResourcesClientFactory sets a ResourcesClient factory in frontend
+func WithResourcesClientFactory(factory resourcesClientFactory) Option {
+	return func(f *frontend) {
+		f.resourcesClientFactory = factory
+	}
+}
+
+// WithVMClientFactory sets a VirtualMachinesClient factory in frontend
+func WithVMClientFactory(factory vmClientFactory) Option {
+	return func(f *frontend) {
+		f.vmClientFactory = factory
+	}
+}
+
+// WithVNetClientFactory sets a VirtualNetworksClient factory in frontend
+func WithVNetClientFactory(factory vNetClientFactory) Option {
+	return func(f *frontend) {
+		f.vNetClientFactory = factory
+	}
+}
+
+// WithAccountsClientFactory sets an AccountsClient factory in frontend
+func WithAccountsClientFactory(factory accountsClientFactory) Option {
+	return func(f *frontend) {
+		f.accountsClientFactory = factory
+	}
+}
+
+// WithKubernetesClientFactory sets a Kubernetes Clientset factory in frontend
+func WithKubernetesClientFactory(factory kubernetesClientFactory) Option {
+	return func(f *frontend) {
+		f.kubernetesClientFactory = factory
+	}
+}
+
+func (f *frontend) KubeActionsFactory() kubeActionsFactory {
+	return f.kubeActionsFactory
+}
+func (f *frontend) ResourcesClientFactory() resourcesClientFactory {
+	return f.resourcesClientFactory
+}
+func (f *frontend) VMClientFactory() vmClientFactory {
+	return f.vmClientFactory
+}
+func (f *frontend) VNetClientFactory() vNetClientFactory {
+	return f.vNetClientFactory
+}
+func (f *frontend) AccountsClientFactory() accountsClientFactory {
+	return f.accountsClientFactory
+}
+func (f *frontend) KubernetesClientFactory() kubernetesClientFactory {
+	return f.kubernetesClientFactory
+}
```

Comment:
Are these "Gets" necessary?  I don't see them being called.

Reply:
They are currently called in `pkg/frontend/admin_openshiftcluster_resources_list.go`, but you're right: they are pointless, since we never access the factories from outside this package. I'll remove them.